src/Sarif.Viewer.VisualStudio/ErrorListService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Driver.Sdk;
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+
+namespace SarifViewer
+{
+    public class ErrorListService
+    {
+        public static ErrorListService Instance = new ErrorListService();
+
+        public static void ProcessSarifLog(ResultLog resultLog)
+        {
+            foreach (RunLog runLog in resultLog.RunLogs)
+            {
+                Instance.WriteRunLogToErrorList(runLog);
+            }
+
+            SarifTableDataSource.Instance.BringToFront();
+        }
+
+        private ErrorListService()
+        {
+            this.documentToLineIndexMap = new Dictionary<string, NewLineIndex>();
+        }
+
+        private Dictionary<string, NewLineIndex> documentToLineIndexMap;
+
+        private IRuleDescriptor GetRule(RunLog runLog, string ruleId)
+        {
+            if (runLog.RuleInfo == null)
+            {
+                return null;
+            }
+
+            foreach (IRuleDescriptor ruleDescriptor in runLog.RuleInfo)
+            {
+                if (ruleDescriptor.Id == ruleId) { return ruleDescriptor; }
+            }
+
+            throw new InvalidOperationException();
```

Comment:
This is a good point. We don't want to allow this case, though. If you provide ruleInfo, that data should be a superset of everything required to create all required formattedMessages. Otherwise, your log file is broken (a viewer can't build your message). 

You might be saying that this code isn't the place to perform validation of this kind, which would make sense.

I definitely think we should address this in the spec.